homeassistant/components/cast/media_player.py

```diff
@@ -103,13 +108,15 @@ def _fill_out_missing_chromecast_info(info: ChromecastInfo) -> ChromecastInfo:
     # Fill out missing information via HTTP dial.
     from pychromecast import dial
 
-    http_device_status = dial.get_device_status(info.host)
+    http_device_status = dial.get_device_status(
+        info.host, services=[info.service],
+        zconf=ChromeCastZeroconf.get_zeroconf())
     if http_device_status is None:
         # HTTP dial didn't give us any new information.
         return info
 
     return ChromecastInfo(
-        host=info.host, port=info.port,
+        service=info.service, host=info.host, port=info.port,
```

Comment:
Should we still keep track of `host` and `port` ?

Reply:
Yeah, unless we can drop support for manually configured casts?

Reply:
I would love to, but I remember the one time we accidentally broke it, people complained, so it is used. I would love to be able to have pychromecast be fully service based and if people do manual configuration, we just convert host to a service.